sdk/core/core-amqp/src/retry.ts

```diff
@@ -120,29 +120,24 @@ async function checkNetworkConnection(host: string): Promise<boolean> {
  */
 export async function retry<T>(config: RetryConfig<T>): Promise<T> {
   validateRetryConfig(config);
-  if (config.times == undefined) config.times = defaultRetryAttempts;
+  if (config.maxRetries == undefined) config.maxRetries = defaultMaxRetries;
   if (config.delayInSeconds == undefined) {
     config.delayInSeconds = defaultDelayBetweenRetriesInSeconds;
   }
   let lastError: MessagingError | undefined;
   let result: any;
   let success = false;
-  for (let i = 0; i < config.times; i++) {
-    const j = i + 1;
-    log.retry(
-      "[%s] Retry for '%s', attempt number: %d",
-      config.connectionId,
-      config.operationType,
-      j
-    );
+  const totalNumberOfAttempts = config.maxRetries + 1;
```

Comment:
I think we shouldn't be changing anything in the `retry` method except `times` -> `maxRetries `. This method should take total number of retry attempts.

I think it's a caller responsibility to pass the correct number of `maxRetries `. In case of `send()` and `receiveBatch()` of Event Hubs, we should increment the maxRetries by 1 before passing the retryConfig to the `retry` method

Reply:
Please see https://github.com/Azure/azure-sdk-for-js/issues/4038#issuecomment-507890491
We decided to not include initial attempt when supplying value of `maxRetries`.

Sorry I thought I commented about implementation decision as well - please see https://github.com/Azure/azure-sdk-for-js/issues/4038#issuecomment-508195565

Reply:
@ramya-rao-a Could you please clarify this?

Reply:
@ShivangiReja I like this model better. In what you and I discussed offline, we landed on doing the +1 on the caller's side to reduce the amount of code changes inside `retry()` in amqp-common assuming that the change is not trivial.

But with @ramya0820's changes here, the change was not too invasive and has the added benefit that all parties involved deal with only `maxRetries` and that the +1 is an internal implementation.

The only thing we need now is for the doc comments for the `retry` method to be very clear on how it uses the `maxRetries`